lib/recurly/pricing/subscription/index.js

```diff
@@ -84,25 +86,21 @@ export default class SubscriptionPricing extends Pricing {
     return new PricingPromise((resolve, reject) => {
       if (currentPlan && currentPlan.code === planCode) {
         currentPlan.quantity = quantity;
-        return resolve(currentPlan);
+        return resolve(clone(currentPlan));
```

Comment:
Why is it necessary to make a deep copy here?

Reply:
If the object returned is a reference to the currentPlan, it can be mutated in the caller and affect the original object used by the SubscriptionPricing instance.